src/App/Repository/EloquentRepository.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/**
+ * Ushahidi Repository
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Repository;
+
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\Usecase;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Support\Collection;
+
+abstract class EloquentRepository implements
+    Usecase\CreateRepository,
+    Usecase\ReadRepository,
+    Usecase\UpdateRepository,
+    Usecase\DeleteRepository,
+    Usecase\ImportRepository
+{
+    protected $connection;
+
+    public function __construct(ConnectionInterface $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * Get the entity for this repository.
+     * @param  Array  $data
+     * @return Ushahidi\Core\Entity
+     */
+    abstract public function getEntity(array $data = null);
+
+    /**
+     * Converts an array/collection of results into an collection
+     * of entities, indexed by the entity id.
+     *
+     * Included directly instead of using Ushahidi\Core\Traits\CollectionLoader
+     * because this implementation returns an Illuminate\Support\Collection
+     *
+     * @param  Array|Iterable $results
+     * @return \Illuminate\Support\Collection
+     */
+    protected function getCollection($results)
```

Comment:
Added this directly to this repo, rather than using the existing trait. Laravel Collections are much more useful than arrays.